swagger_server/test/test_distance_controller.py

```diff
@@ -33,39 +34,49 @@ def test_non_existing_sample(self, sub_type, sample):
         )
 
     def test_nearest_leaf(self):
-        response = self.request(self.node.name, 'nearest-leaf-node')
+        response = self.request(self.node['name'], 'nearest-leaf-node')
 
         self.assert200(response)
         self.assertDictEqual(
             json.loads(response.data.decode()),
             {
-                'leaf_id': self.leaf.name,
+                'leaf_id': self.leaf['name'],
                 'distance': self.dist
             }
         )
 
     def test_nearest_neighbors(self):
-        response = self.request(self.node.name, 'nearest-neighbours')
+        response = self.request(self.node['name'], 'nearest-neighbours')
 
         self.assert200(response)
 
         actual = json.loads(response.data.decode())
         expected = [{
-            'experiment_id': n.name,
-            'distance': self.node.neighbors.relationship(n).dist
-        } for n in self.neighbours]
+            'experiment_id': n['name'],
+            'distance': d
+        } for n, d in zip(self.neighbours, self.neighbour_dists)]
 
         for a in actual:
             self.assertIn(a, expected)
 
+    def test_no_neighbor(self):
+        response = self.request(self.isolated_node_name, 'nearest-neighbours')
+
+        self.assert200(response)
+
+        actual = json.loads(response.data.decode())
+        expected = []
+        assert actual == expected
+
```

Comment:
Please also test no "nearest leaf" case.

Reply:
I thought all nodes must have a lineage?

Reply:
As mentioned above, we did not make this constraint specifically for this API.

Reply:
But at the conceptual level, we always ensure all samples have their lineage, so would such a test redundant?